rfc/rfc-0000-move-off-monorepo.md

```diff
@@ -0,0 +1,57 @@
+
+Adopting a microservice approach for Cartographer will aid extensions and help build up the wider ecosystem and community around it.
+This RFC proposes how we will adopt this microservice style approach that will help grow the Cartographer ecosystem, encourage new integrations and extensions is a sustainable and maintainable way that suits the project and its aspirations.
+
+## Motivation
+
+The motivation behind Cartographer as an OSS project aims to build a community and ecosystem to aid supply chain choreography.  Cartographer itself is an "umbrella" for this ecosystem, the first step of this was to build the core components and now we are looking ahead to extend by adding more capabilities.
+ 
+As expressed by maintainers of the https://github.com/vmware-tanzu/cartographer repository it will become hard to manage it if we use it as a monorepo and continue to add more capabilities like conventions etc.
+It is expected that this RFC describes how we split the current repository so it allows the Cartographer OSS project to evolve and scale beyond the concrete capabilities the core aims to provide.
+For example splitting out from the current repo:
+
+-  the __`Runnable`__ feature so it can release independently and install standalone in a target deployment cluster,  avoids deploying all of Cartographer which contains build cluster CRDs not desired in runtime clusters
+- the __examples directory__ so we can build up a wide range of examples that include integrations with wider platform services like conventions.  This also helps encourage contribution of community supply chains for different integrations where we can add automated testing that runs against different setups.
+- the __website__ so we can add documentation for these new integrations and platform capabilities outside of the core repository.  I.e. Documenting services like the conventions or a Jenkins integration should not require a change to the core cartographer repository, this should live outside in its own repository.
+
+## Possible Solutions
+
+There seems to be only two options possible solutions:
+- __monorepo__ we could revisit the [convention proposal](https://github.com/vmware-tanzu/cartographer/pull/514) decision and look at ways to address the monorepo concerns however there seemed a general consensus on the [office hours call](https://www.youtube.com/watch?v=iCURQqV52Uw&t=1173s) that this was not desired.
+- __microservice__ As conventions are now being worked on in a different repository, we should look at splitting out parts that currently reside in https://github.com/vmware-tanzu/cartographer.
+
+Suggestions:
+
+-  __`Runnable`__ feature so it can install standalone in a target deployment cluster and avoids deploying all of Cartographer
+-  __examples directory__ so we can build up a wide range of examples that include integrations with wider platform services like conventions.
```

Comment:
There's been some feedback that a community maintained set of examples that live on their own repo would aid contributions for wider integrations and scenarios.  @vrabbi did you have some thoughts on this?

Once we add conventions and say Jenkins plus a growing number of other integrations it would be good if they existed in a separately maintained repo we can add more automation beyond the scope of the core cartographer repo.

Reply:
I think that having an examples repo seperate is of huge value. I domt think e2e needs to be run against all examples so having core functionailty for testing in the main repo makes sense but by having an additional repo only for examples, it can server as a central location for examples of using different integrations like kaniko,jenkins,argo etc.

Reply:
And having that under the cartographer umbrella is key i believe from a community perspective to show that cartographer truly is pluggable by design. If its in an official cartographer repo it gains more credability.